File: Elixir/hello_world_again/lib/k03/k03.ex
defmodule K03 do
  defp question_1 do
    age = IO.gets("年齢を入力してください。＞") |> String.trim() |> String.to_integer()
    if age < 20 do
      IO.puts("未成年なので購入できません。")
    end
  end

  defp question_2 do
    height = IO.gets("身長を入力してください。＞")
    |> String.trim()
    |> String.to_float()
    height = height / 100.0

    weight = IO.gets("体重を入力してください。＞")
    |> String.trim()
    |> String.to_float()

    standard = height * height * 22.0
    IO.puts("あなたの標準体重は#{standard}です。")

    cond do
      weight > standard && (weight - standard) / standard * 100.0 > 14.0 ->
        IO.puts("太り気味です。")
      weight < standard && (weight - standard) / standard * 100.0 < -14.0 ->
        IO.puts("痩せ気味です。")
      true -> IO.puts("普通ですね。")
    end
  end

  defp question_3 do
    random_number = Enum.random(0..99)
    IO.puts("０から９９の範囲の数値が決定されました。")
    guess = IO.gets("決められた数値を予想し、この数値よりも大きな値を入力してください＞")
    |> String.trim()
    |> String.to_integer()

    IO.puts("決められた数値は#{random_number}です。")
    IO.puts(if guess > random_number do
      "正解です。"
    else
      "不正解です。"
    end)
  end

  defp question_4 do
    random_number = Enum.random(0..99)
    IO.puts("０から９９の範囲の数値が決定されました。")
    guess = IO.gets("決められた数値を予想し、この数値よりも大きな値を入力してください＞")
    |> String.trim()
    |> String.to_integer()

    IO.puts("決められた数値は#{random_number}です。")
    IO.puts(cond do
      guess < 0 || guess > 99 -> "反則です！"
      guess > random_number && guess - random_number <= 10 -> "大正解です！"
      guess < random_number && random_number - guess <= 10 -> "惜しい！"
      guess == random_number -> "お見事！"
      true -> if guess > random_number do
        "正解です。"
      else
        "不正解です。"
      end
    end)
  end

  @spec execute(integer()) :: :noop | :ok
  def execute(num) do
    case num do
      1 -> question_1()
      2 -> question_2()
      3 -> question_3()
      4 -> question_4()
      _ -> :noop
    end
  end
end

File: PHP/hello_world_again/src/K03.php
<?php

class K03 implements IExecutable
{

    public function question_1()
    {
        $age = readline("年齢を入力してください。＞");
        if ($age < 20)
        {
            echo "未成年なので購入できません。";
        }
    }

    public function question_2()
    {
        $height = readline("身長を入力してください。＞");
        $height /= 100.0;
        $weight = readline("体重を入力してください。＞");
        $standard = ($height * $height) * 22.0;
        echo "あなたの標準体重は" . $standard . "です。\n";

        if ($weight > $standard && ($weight - $standard) / $standard * 100.0 > 14.0)
        {
            echo "太り気味です。";
        }
        else if ($weight < $standard && ($weight - $standard) / $standard * 100.0 < -14.0)
        {
            echo "痩せ気味です。";
        }
        else
        {
            echo "普通ですね。";
        }
    }

    public function question_3()
    {
        $random_number = rand(0, 99);
        echo "０から９９の範囲の数値が決定されました。\n";
        $guess = readline("決められた数値を予想し、この数値よりも大きな値を入力してください＞");
        echo "決められた数値は" . $random_number . "です。\n";
        if ($guess > $random_number)
        {
            echo "正解です。";
        }
        else
        {
            echo "不正解です。";
        }
    }

    public function question_4()
    {
        $random_number = rand(0, 99);
        echo "０から９９の範囲の数値が決定されました。\n";
        $guess = readline("決められた数値を予想し、この数値よりも大きな値を入力してください＞");
        echo "決められた数値は" . $random_number . "です。\n";

        if ($guess < 0 || $guess > 99)
        {
            echo "反則です！";
        }
        else if ($guess > $random_number && $guess - $random_number <= 10)
        {
            echo "大正解です！";
        }
        else if ($random_number > $guess && $random_number - $guess <= 10)
        {
            echo "惜しい！";
        }
        else if ((int)$guess === $random_number)
        {
            echo "お見事！";

        }
        else
        {
            if ($guess > $random_number)
            {
                echo "正解です。";
            }
            else
            {
                echo "不正解です。";
            }
        }
    }

    public function execute(int $num): void
    {
        switch ($num) {
            case 1:
                $this->question_1();
                break;
            case 2:
                $this->question_2();
                break;
            case 3:
                $this->question_3();
                break;
            case 4:
                $this->question_4();
                break;
        }
    }
}